# HG changeset patch
# User Rik <rik@octave.org>
# Date 1459311016 25200
#      Tue Mar 29 21:10:16 2016 -0700
# Node ID 96c768d898f0a4a826fe16ce6e04988cc1d76983
# Parent  6619945e4434c94b4f8ed93fa7860159e4f86544
Use stricter input validation for ezplot N, DOM inputs (bug #46952).

* __ezplot__.m: Only allow DOM or N to be set once.

diff --git a/scripts/plot/draw/private/__ezplot__.m b/scripts/plot/draw/private/__ezplot__.m
--- a/scripts/plot/draw/private/__ezplot__.m
+++ b/scripts/plot/draw/private/__ezplot__.m
@@ -245,24 +245,24 @@ function [h, needusage] = __ezplot__ (pl
     iarg = 2;
   endif
   while (iarg <= nargin)
     arg = varargin{iarg++};
     if (ischar (arg) && strcmp (arg, "circ"))
       circ = true;
     elseif (ischar (arg) && strcmp (arg, "animate"))
       animate = true;
-    elseif (isscalar (arg))
+    elseif (isscalar (arg) && (n == 60 || n == 500))
       n = arg;
-    elseif (numel (arg) == 2)
+    elseif (numel (arg) == 2 && isempty (domain))
       domain = [arg(1) arg(2) arg(1) arg(2)];
-    elseif (numel (arg) == 4)
+    elseif (numel (arg) == 4 && isempty (domain))
       domain = arg(:).';
     else
-      error ("%s: expecting scalar N or 2-, 4-element vector DOM", ezfunc);
+      error ("%s: expecting scalar N, or 2-/4-element vector DOM", ezfunc);
     endif
   endwhile
 
   if (circ && (iscontour || isplot3 || isplot))
     needusage = true;
     return;
   elseif (circ && parametric)
     error ("%s: can not have both circular domain and parametric function",
